### PR TITLE
[Quest API] Add HasBotSpellEntry() to Perl/Lua.

### PR DESCRIPTION
**Pull request by Aeadoin**
Add Perl & Lua method to validate if a Spell ID is in the Bot Spells Entries based on the Bot Class.

Perl: $bot->CheckBotSpellEntries;
Lua: bot:CheckBotSpellEntries;

### CODE REVIEW

File: zone/lua_bot.cpp [hunk 1]
```diff
@@ -109,29 +109,35 @@ void Lua_Bot::SetExpansionBitmask(int expansion_bitmask, bool save) {
...
+	.def("GetBotItemIDBySlot", (uint32(Lua_Bot::*)(uint16)) & Lua_Bot::GetBotItemIDBySlot)
+	.def("GetExpansionBitmask", (int(Lua_Bot::*)(void)) & Lua_Bot::GetExpansionBitmask)
+	.def("GetOwner", (Lua_Mob(Lua_Bot::*)(void)) & Lua_Bot::GetOwner)
+	.def("HasBotItem", (bool(Lua_Bot::*)(uint32)) & Lua_Bot::HasBotItem)
+	.def("OwnerMessage", (void(Lua_Bot::*)(std::string)) & Lua_Bot::OwnerMessage)
+	.def("RemoveBotItem", (void(Lua_Bot::*)(uint32)) & Lua_Bot::RemoveBotItem)
+	.def("SetExpansionBitmask", (void(Lua_Bot::*)(int)) & Lua_Bot::SetExpansionBitmask)
+	.def("SetExpansionBitmask", (void(Lua_Bot::*)(int, bool)) & Lua_Bot::SetExpansionBitmask)
+	.def("SignalBot", (void(Lua_Bot::*)(int)) & Lua_Bot::SignalBot)
+	.def("HasBotSpellEntry", (bool(Lua_Bot::*)(uint16)) & Lua_Bot::HasBotSpellEntry);
```
Comment (Kinglykrab) [inline zone/lua_bot.cpp:140]: Ordering.

File: zone/perl_bot.cpp [hunk 2]
```diff
@@ -132,6 +137,7 @@ void perl_register_bot()
 	package.add("SetExpansionBitmask", (void(*)(Bot*, int))&Perl_Bot_SetExpansionBitmask);
 	package.add("SetExpansionBitmask", (void(*)(Bot*, int, bool))&Perl_Bot_SetExpansionBitmask);
 	package.add("SignalBot", &Perl_Bot_SignalBot);
+	package.add("HasBotSpellEntry", &Perl_Bot_HasBotSpellEntry);
```
Comment (Kinglykrab) [inline zone/perl_bot.cpp:140]: Ordering.

File: zone/botspellsai.cpp [hunk 3]
```diff
@@ -3185,4 +3185,49 @@ void Bot::AI_Bot_Event_SpellCastFinished(bool iCastSucceeded, uint16 slot) {
...
+		debug_msg.append(" (not found)");
+	}
+
+	if (spell_list) {
+		for (const auto& iter : spell_list->entries) {
+			LogAIDetail("([{}]) [{}]", iter.spellid, spells[iter.spellid].name);
+		}
+	}
+
+	LogAIModerate("fin (spell list)");
```
Comment (Kinglykrab) [inline zone/botspellsai.cpp:3221]: Not sure how necessary this is here.

**Comment by Kinglykrab** [comment]
Few comments, other than that, looks good.

File: zone/botspellsai.cpp (hunk 3 above)
Comment (Aeadoin) [reply zone/botspellsai.cpp]: Removed it.

File: zone/perl_bot.cpp (hunk 2 above)
Comment (Aeadoin) [reply zone/perl_bot.cpp]: Fixed

File: zone/lua_bot.cpp (hunk 1 above)
Comment (Aeadoin) [reply zone/lua_bot.cpp]: Fixed

**Review by Kinglykrab — APPROVED** [review]

**Pull request merged**